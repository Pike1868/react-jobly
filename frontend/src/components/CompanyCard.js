import React, { useEffect, useState } from "react";
import Button from "@mui/material/Button";
import Card from "@mui/material/Card";
import CardActions from "@mui/material/CardActions";
import CardContent from "@mui/material/CardContent";
import CardMedia from "@mui/material/CardMedia";
import Grid from "@mui/material/Grid";
import Typography from "@mui/material/Typography";
import { Link } from "react-router-dom";

export default function CompanyCard({ company }) {
  const { handle, name, description, numEmployees } = company;
  return (
    <Grid item xs={12} sm={6} md={4}>
      <Card
        sx={{
          height: "100%",
          display: "flex",
          flexDirection: "column",
        }}
      >
        <CardMedia
          component="div"
          sx={{
            pt: "56.25%",
          }}
          image={"https://source.unsplash.com/random?backgrounds"}
        />
        <CardContent sx={{ flexGrow: 1 }}>
          <Typography gutterBottom variant="h5" component="h2">
            {name}
          </Typography>
          <Typography align="left">{description}</Typography>
          <Typography variant="body2" align="left" sx={{ padding: "16px 0 0" }}>
            Employees: {numEmployees}
          </Typography>
        </CardContent>

        <CardActions>
          <Link to={`/companies/${handle}`} style={{ textDecoration: "none" }}>
            <Button size="small">View Company</Button>
          </Link>
        </CardActions>
      </Card>
    </Grid>
  );
}
