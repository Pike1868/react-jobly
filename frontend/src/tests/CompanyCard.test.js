import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CompanyCard from "../components/CompanyCard";

describe("CompanyCard Component", () => {
  const mockCompany = {
    handle: "mock-company",
    name: "Mock Company",
    description: "Mock company description.",
    numEmployees: 500,
  };

  it("renders without crashing", () => {
    render(
      <MemoryRouter>
        <CompanyCard company={mockCompany} />
      </MemoryRouter>
    );
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CompanyCard company={mockCompany} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
