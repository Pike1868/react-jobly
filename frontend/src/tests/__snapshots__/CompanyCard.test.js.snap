// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyCard Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-13i4rnv-MuiGrid-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-16vq84g-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardMedia-root css-1kxdiu9-MuiCardMedia-root"
        role="img"
        style="background-image: url(https://source.unsplash.com/random?backgrounds);"
      />
      <div
        class="MuiCardContent-root css-2r3irp-MuiCardContent-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
        >
          Mock Company
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignLeft css-zpphrz-MuiTypography-root"
        >
          Mock company description.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-alignLeft css-nitxol-MuiTypography-root"
        >
          Employees: 500
        </p>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-dnrpxu-MuiCardActions-root"
      >
        <a
          href="/companies/mock-company"
          style="text-decoration: none;"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-1knaqv7-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            View Company
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
